test/lib.bash

```diff
@@ -73,34 +76,52 @@ function run_knative_serving_tests {
   # Adding scc for anyuid to test TestShouldRunAsUserContainerDefault.
   oc adm policy add-scc-to-user anyuid -z default -n serving-tests
 
-  local failed=0
-  image_template="registry.svc.ci.openshift.org/openshift/knative-${knative_version}:knative-serving-test-{{.Name}}"
   export GATEWAY_NAMESPACE_OVERRIDE="knative-serving-ingress"
+}
 
-  # Rolling upgrade tests must run first because they upgrade Serverless to the latest version
-  if [[ $RUN_KNATIVE_SERVING_UPGRADE_TESTS == true ]]; then
-     run_knative_serving_rolling_upgrade_tests || failed=1
-  fi
+function run_knative_serving_e2e_and_conformance_tests {
+  logger.info "Running E2E and conformance tests"
+  (
+  local knative_version=$1
 
-  if [[ $RUN_KNATIVE_SERVING_E2E == true ]]; then
-    run_knative_serving_e2e_and_conformance_tests || failed=1
+  if [[ -z ${KNATIVE_SERVING_HOME+x} ]]; then
+    checkout_knative_serving "$knative_version"
   fi
+  cd "$KNATIVE_SERVING_HOME" || return $?
 
-  rm -rf "$tmp_gopath"
-  return $failed
-  )
-}
+  prepare_knative_serving_tests || return $?
+
+  local failed=0
+  image_template="registry.svc.ci.openshift.org/openshift/knative-${knative_version}:knative-serving-test-{{.Name}}"
 
-function run_knative_serving_e2e_and_conformance_tests {
-  logger.info "Running E2E and conformance tests"
   go test -v -tags=e2e -count=1 -timeout=30m -parallel=3 ./test/e2e ./test/conformance/... \
     --resolvabledomain --kubeconfig "$KUBECONFIG" \
-    --imagetemplate "$image_template" || return 1
+    --imagetemplate "$image_template" || failed=1
+
+  if [[ "$GOPATH" =~ .*gopath-[0-9a-zA-Z]{10} ]]; then
```

Comment:
+1

Reply:
Actually, in my latest commit I don't remove the folder at all. We might need it later for extracting JUnit XML files or something else. I think it's not necessary to remove it in CI. It will be clean anyway when CI ends.

Reply:
Sure. But, for running locally it will be nice to remove that junk.